lib/datasource/go/index.ts

```diff
@@ -88,6 +89,35 @@ export async function getPkgReleases({
   const source = await getDatasource(lookupName);
   if (source && source.datasource === github.id) {
     const res = await github.getPkgReleases(source);
+
+    /**
+     * github.com/org/mod/submodule should be tagged as submodule/va.b.c
+     * and that tag should be used instead of just va.b.c, although for compatibility
+     * the old behaviour stays the same.
+     */
+    const nameParts = lookupName.split('/');
+    logger.trace(`go.getPkgReleases.nameParts:${nameParts}`);
+    logger.trace(`go.getPkgReleases.releases:${JSON.stringify(res.releases)}`);
+    if (nameParts.length > 3) {
+      const prefix = nameParts.slice(3, nameParts.length).join('/');
+      logger.trace(`go.getPkgReleases.prefix:${prefix}`);
+      const submodReleases = res.releases
+        .filter(
+          release => release.version && release.version.startsWith(prefix)
+        )
+        .map(release => {
+          const r2 = release;
+          r2.version = r2.version.replace(`${prefix}/`, '');
+          return r2;
+        });
+      logger.trace(
+        `go.getPkgReleases.filteredReleases:${JSON.stringify(submodReleases)}`
+      );
```

Comment:
```suggestion
```

Reply:
I think these are traces that would help debug. Do you really want me to remove it?

Reply:
updated suggestion